nbsafety/safety.py

```diff
@@ -413,6 +413,9 @@ def get_cell_dependencies(self, cell_num: int) -> Dict[int, str]:
             - dict (int, str): map from required cell number to code
                 representing dependencies
         """
+        if cell_num not in self.cell_content_by_counter.keys():
```

Comment:
Maybe `if cell_num <= self._cell_counter`? Actually not sure if `<` or `<=` -- what you have might be safe.